### PR TITLE
Async dbus interface

### PR DESCRIPTION
**Pull request by NeroReflex**
This makes GPU dbus interface fully async and therefore allows usage of async functions inside dbus logic without crashing the runtime.

This is needed since lots of crates are async and the future asus work will need this.

This PR is a rework of my in-development asus-specific device and needs testing before merging.

### CODE REVIEW

File: src/performance/gpu/amd/asus.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
+use std::sync::Arc;
```
Comment (pastaq) [inline src/performance/gpu/amd/asus.rs:1]: Is this file necessary for this PR? It doesn't appear to be used yet and most of the branches are incomplete anyway. It also seems to be out of scope for the PR subject.

**Review by pastaq — CHANGES_REQUESTED** [review]
Everything seems to work from my testing. I'd only request that all files remain in scope of the PR.

File: src/performance/gpu/amd/asus.rs (hunk 1 above)
Comment (NeroReflex) [reply src/performance/gpu/amd/asus.rs:1]: This file exists since the previous PR (as a placeholder, actually unused) and so I had to edit it to met the new interface.

I tried to minimize modification on this file, but that line could have sneaked in.

I will see if removing such line makes the build success.

Comment (pastaq) [reply src/performance/gpu/amd/asus.rs:1]: Oh, I guess I missed that in the last PR. I suppose its fine then since you're just aligning everything to async and the calls to this file are commented out. In the future please try to be a good citizen of git and keep all changes related to the goal a specific PR, it makes reverting breaking changes and managing merge conflicts easier in the future.

**Review by pastaq — APPROVED** [review]

File: src/performance/gpu/amd/asus.rs (hunk 1 above)
Comment (NeroReflex) [reply src/performance/gpu/amd/asus.rs:1]: I am sorry, those sneaked in in the previous PR. I had them initially to test that the mechanism of specializing the hardware was working and forgot to remove them. I am still in time to remove all of those just to re-add them later.
